web/src/components/predictions/predictions-game.js

```diff
@@ -1,21 +1,289 @@
+  const [borderColor, setBorderColor] = useState();
+  const [borderWidth, setBorderWidth] = useState(1);
+
+  // add user to firebase if doesn't exist
+  if (snapshot && !snapshot.exists()) {
+    if (!questionsLoading) {
+      let initial = {};
+      let initialScore = {};
+      initial["score"] = {};
+      questions.forEach((question) => (initial["score"][question.key] = 0));
+      initialScore["nickname"] = user.displayName;
+      initialScore["score"] = 0;
+      firebase
+        .database()
+        .ref("predictions_users/" + user.uid)
+        .set(initial);
+      firebase
+          .database()
+          .ref("predictions/leaderboard/" + user.uid)
+          .update(initialScore);
+    }
+  }
+
+  // answer is the index of the choice for MC questions, or the correct number for range questions
+  function calculateScore(isMC, prediction, answer, qid, uid, range) {
+    let score = 0;
+    const scale = 60;
+    let total = 0;
+    let updates = {};
+    let leaderboardUpdates = {};
+
+    if (answer && prediction) {
+      if (isMC) {
+        if (prediction.length === 1) {
+          // binary scoring
+          if (answer) {
+            score = (1 - prediction[0] / 100) ** 2 * 2;
+          } else {
+            score = (prediction[0] / 100) ** 2 * 2;
+          }
+          score = -score * scale + scale / 2;
+        } else {
+          // multi-category scoring
+          for (let i = 0; i < prediction.length; i++) {
+            if (i === answer) {
+              score += (1 - prediction[i] / 100) ** 2;
+            } else {
+              score += (prediction[i] / 100) ** 2;
+            }
+          }
+          score =
+            -score * scale +
+            (scale * (prediction.length - 1)) / prediction.length;
+        }
+      } else {
+        // range-based scoring
+        if (answer >= prediction[0] && answer <= prediction[1]) {
+          score = scale * (1 - (prediction[1] - prediction[0]) / (range[1] - range[0]));
+        }
+        else {
+            score = -scale * (1 - (prediction[1] - prediction[0]) / (range[1] - range[0]));
+        }
+      }
+      // update firebase with score
+      updates[qid] = score;
+      firebase
+        .database()
+        .ref("predictions_users/" + uid + "/score")
+        .update(updates)
+        .then(() => {
+          total = Object.values(snapshot.child("score").val()).reduce((a, b) => a + b, 0);
+          leaderboardUpdates["score"] = total;
+          firebase
+              .database()
+              .ref("predictions/leaderboard/" + uid)
+              .update(leaderboardUpdates);
+        });
+    }
+    return score;
+  }
 
-  const afterSubmission = (event) => {
-    event.preventDefault();
+  function displayScore(score, explanation) {
+    return (
+      <p>
+        <Text> {explanation} </Text>
+        <Text>
+          You received <strong>{score ? score.toFixed(2) : 0}</strong> points
+          for this prediction.
+        </Text>
+      </p>
```

Comment:
Is there a better way to do this than using `<Text>` tags inside a `<p>` tag? Maybe using a `<div>` or something with padding?